### PR TITLE
Add NavBar

### PR DESCRIPTION
**Pull request by barshathakuri**
## Addresses:
- #2 

## Changes
- Add Nav Bar in Home Page

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,7 +6,7 @@ yarn-debug.log*
 yarn-error.log*
 pnpm-debug.log*
 lerna-debug.log*
-
+ui
```
Comment (samshara) [inline .gitignore]: Let's delete the folder completely. Let's remove this as well from here.

File: src/hooks/useBooleanState.ts
Comment (samshara) [inline src/hooks/useBooleanState.ts:1]: we could remove this file entirely and import this hook from @ifrc-go/ui

File: tsconfig.json [hunk 2]
```diff
@@ -9,7 +9,8 @@
             "#hooks/*": ["./src/hooks/*"],
             "#strings/*": ["./src/strings/*"],
             "#utils/*": ["./src/utils/*"],
-            "#views/*": ["./src/views/*"]
+            "#views/*": ["./src/views/*"],
+            "#ui/*": ["./src/ui/src/components/*"],
```
Comment (samshara) [inline tsconfig.json]: Let's remove this

**Review by samshara — APPROVED** [review]

**Review by subinasr — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -70,23 +71,24 @@ jobs:
...
+  #     - name: Install dependencies
+  #       run: yarn install
 
-      - name: Typecheck
-        run: yarn typecheck
+  #     - name: Typecheck
+  #       run: yarn typecheck
   build:
     name: Build
+    # needs: [lint, css-lint, typecheck, test]
```
Comment (tnagorra) [inline .github/workflows/ci.yml]: Any reason to comment this out?

File: package.json [hunk 4]
```diff
@@ -21,21 +21,24 @@
...
+        "@ifrc-go/ui": "^1.0.0",
+        "@mapbox/mapbox-gl-draw": "^1.4.3",
+        "@sentry/react": "^7.81.1",
         "@togglecorp/fujs": "^2.1.1",
-        "@togglecorp/re-map": "^0.2.0-beta-6",
+        "@togglecorp/re-map": "^0.1.4",
+        "@turf/bbox": "^6.5.0",
         "graphql": "^16.8.1",
-        "mapbox-gl": "^1.13.0",
+        "mapbox-gl": "^3.2.0",
```
Comment (tnagorra) [inline package.json]: We cannot use mapbox-gl 3

File: tsconfig.json [hunk 5]
```diff
@@ -1,15 +1,13 @@
 {
     "compilerOptions": {
         "paths": {
-            "#generated/*": ["./generated/*"],
             "#assets/*": ["./src/assets/*"],
```
Comment (tnagorra) [inline tsconfig.json]: Any reason to remove this alias?

File: src/views/AlertMap/index.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+            withHeaderBorder
+            childrenContainerClassName={styles.content}
+            actions={(
+                <Link
+                    className={styles.sources}
+                    to="/"
+                >
+                    {strings.mapViewAllSources}
+                </Link>
+            )}
```
Comment (tnagorra) [inline src/views/AlertMap/index.tsx:48]: Is the link correct?

File: src/utils/map.ts [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+    return {
+        'circle-color': color,
+        'circle-radius': sizeMap[size] ?? DEFAULT_CIRCLE_SIZE,
+        'circle-opacity': opacityMap[opacity] ?? DEFAULT_CIRCLE_OPACITY,
+        'circle-pitch-alignment': 'map',
+    };
+}
+
+export function getPointCircleHaloPaint(
```
Comment (tnagorra) [inline src/utils/map.ts]: Are we using this? Let's remove all functions that are not used

File: src/utils/constants.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
+export const defaultChartMargin = {
```
Comment (tnagorra) [inline src/utils/constants.ts]: Let's remove all constants that are not used or will not be used.

File: src/components/Navbar/index.tsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+                <NavigationTabList
+                    className={styles.menuItem}
+                    variant="tertiary"
+                >
+                    <Link
+                        to="/"
+                        className={styles.menuItem}
+                    >
+                        {strings.headerMenuHome}
+                    </Link>
```
Comment (tnagorra) [inline src/components/Navbar/index.tsx:92]: Let's add FIXME on all the Link.
Looks like we are only using placeholder path

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/AlertMap/index.tsx (hunk 6 above)
Comment (barshathakuri) [reply src/views/AlertMap/index.tsx:48]: We will add source page in next phase.

File: package.json [hunk 10]
```diff
@@ -1,82 +1,84 @@
...
-        "unimported": "1.31.1",
-        "vite": "^5.2.6",
-        "vite-plugin-checker": "^0.6.2",
-        "vite-plugin-compression2": "^1.0.0",
-        "vite-plugin-svgr": "^4.2.0",
-        "vite-plugin-webfont-dl": "^3.9.1",
-        "vite-tsconfig-paths": "^4.2.2",
-        "vitest": "^1.1.0"
-    }
+  "name": "ifrc-alert-hub",
```
Comment (tnagorra) [inline package.json]: Let's keep tab width as 4 spaces

File: src/env.d.ts [hunk 11]
```diff
@@ -1,10 +0,0 @@
-/// <reference types="vite/client" />
```
Comment (tnagorra) [inline src/env.d.ts:1]: Any reason we are removing this? This is helpful to make env typesafe.

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/env.d.ts (hunk 11 above)
Comment (barshathakuri) [reply src/env.d.ts:1]: I have moved the env typesafe file inside declarations folder.

**Review by samshara — APPROVED** [review]

File: package.json [hunk 12]
```diff
@@ -33,9 +35,9 @@
     "devDependencies": {
         "@eslint/eslintrc": "^3.0.2",
         "@julr/vite-plugin-validate-env": "^1.0.1",
-        "@types/mapbox-gl": "^1.13.0",
-        "@types/node": "^20.11.30",
-        "@types/react": "^18.2.70",
+        "@types/mapbox-gl": "^3.1.0",
```
Comment (tnagorra) [inline package.json]: We should also downgrade the typings.

File: src/components/GlobalFooter/i18n.json [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    "strings": {
+        "footerAboutAlertHub":"About Alert Hub",
+        "footerAboutAlertHubDesc":"The aim of IFRC Alert Hub is to provide timely and effective emergency alerts to communities worldwide, empowering them to protect their lives and livelihoods.",
+        "footerFindOutMore":"Find out more",
+        "footerHelpfulLinks":"Helpful links",
+        "footerOpenSourceCode":"Open Source Code",
+        "footerApiDocumentation":"API Documentation",
+        "footerOtherResources":"Other Resources",
+        "footerContactUs":"Contact Us",
+        "footerIFRC":"© IFRC {year} v{appVersion}",
```
Comment (tnagorra) [inline src/components/GlobalFooter/i18n.json]:
```suggestion
        "footerIFRC":"© IFRC Alert Hub {year} v{appVersion}",
```

**Review by tnagorra — APPROVED** [review]

**Pull request merged**